### ISSUE TITLE
bbackupd never deleting unused store root folder entries. (Trac #18)

### ISSUE THREAD

**Issue by jamesog** [jamesog/boxbackup#18]
A problem with bbackupd (both Win32 and UN*X): bbackupd will detect unused store root folder entries during a sync run, but will attempt to wait up to 2 days, before marking those for deletion during one of the subsequent sync cycles. The deadline and deletion information is lost across bbackupd runs, effectively never deleting redundant store root folder entries, unless you keep your bbackupd running for at least 48 hours (rarely happens in the laptop/workstation scenario).

Migrated from https://www.boxbackup.org/ticket/18
```json
{
    "status": "closed", 
    "changetime": "2007-07-26T22:03:19", 
    "description": "A problem with bbackupd (both Win32 and UN*X): bbackupd will detect unused store root folder entries during a sync run, but will attempt to wait up to 2 days, before marking those for deletion during one of the subsequent sync cycles. The deadline and deletion information is lost across bbackupd runs, effectively never deleting redundant store root folder entries, unless you keep your bbackupd running for at least 48 hours (rarely happens in the laptop/workstation scenario).", 
    "reporter": "gniemcew", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1185487399000000", 
    "component": "bbackupd", 
    "summary": "bbackupd never deleting unused store root folder entries.", 
    "priority": "normal", 
    "keywords": "folder,unused,delete", 
    "version": "0.10", 
    "time": "2007-01-11T17:56:21", 
    "milestone": "0.11", 
    "owner": "ben", 
    "type": "defect"
}
```

**Comment by jamesog** [comment]
Trac update at 20070111T20:10:48: gniemcew changed attachment from "" to "bbackupd-obj-store-v2.patch"

**Comment by jamesog** [comment]
Trac update at 20070515T20:56:32: chris commented:

> (In [1658]) Apply Gary's patch to save the list of unused root directory entries
> in the store object info file, so that they will persist across restarts
> (#18)

**Comment by jamesog** [comment]
Trac update at 20070515T23:31:10: chris commented:

> (In [1663]) New win32 native build (1662) with the following fixes and new features:
> 
>  * Fix backup, restore and compare of files over 2 GB, hopefully
>    ([1585], [1586], [1587], [1598], [1599], [1621], [1623])
> 
>  * Log progress of large file backups at trace level ([1589], [1590])
> 
>  * Notify (log) during full file uploads ([1594])
> 
>  * Improve error messages in bbackupquery when comparing or getting a file
>    fails with an exception ([1597], [1600])
> 
>  * Improve error reporting in bbackupquery ([1638])
> 
>  * Initialise logging framework and use for banner in bbackupquery
>    ([1601], [1602])
> 
>  * Include LICENSE.txt in binary package, but not Nick's old ReadMe.txt
>    ([1655])
> 
>  * Save unused locations across bbackupd restarts, so that they will be
>    deleted when their time is up, thanks to Gary ([1658], #18)
> 
>  * Respect server hard limits rather than soft limits, to allow use
>    of zero as a soft limit, thanks to Gary ([1659], [1662], #19)
> 
>  * Fix restore to a directory under the root, with or without drive
>    letter (c:\test\ or \test\), thanks to Mikael Syska 
>    ([1660], [1661])

**Comment by jamesog** [comment]
Trac update at 20070517T19:31:39: chris commented:

> (In [1669]) Apply remaining parts of Gary's patch, missing from [1658], thanks Gary
> (#18)

**Comment by jamesog** [comment]
Trac update at 20070518T20:29:49: 
* chris changed status from "new" to "closed"
* chris commented:

> (In [1672]) Fixed object store read/write, fixes #18 properly, thanks Gary!


* chris changed resolution from "" to "fixed"

**Comment by jamesog** [comment]
Trac update at 20070726T21:54:05: chris commented:

> (In [1753]) Apply Gary's patch to save the list of unused root directory entries
> in the store object info file, so that they will persist across restarts.
> (#18, #3, merges [1658])

**Comment by jamesog** [comment]
Trac update at 20070726T22:03:19: chris commented:

> (In [1766]) Fixed object store read/write, fixes #18 properly, thanks Gary!
> (#3, merges [1672])
